src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -234,6 +258,9 @@ trait Metric extends Product with Iterable[(String,String)] {
   /** Gets an iterator over the fields of this metric in the order they were defined.  Returns tuples of names and values */
   override def iterator: Iterator[(String,String)] = this.names.zip(this.values).iterator
 
+  /** The delimiter for collection types. */
+  protected def collectionDelimiter: Char = Metric.DefaultCollectionDelimiter
```

Comment:
Is this so that the delimiter can be overridden by child classes? If so, wouldn't you also want to make this the default for the functions above that perform iteration and reading instead of the static `DefaultCollectionDelimiter`? Is that possible?